src/bitte_ci/libgit2.cr

```diff
@@ -396,6 +432,13 @@ module Git
     checkout_options
   end
 
+  def self.fetch_options_init(username, password)
+    check(LibGit.git_fetch_options_init(out fetch_options, 1))
+    credentials = authenticate(username, password)
+    fetch_options.callbacks.credentials = credentials
```

Comment:
This assignation is obviously not correctly typed. But I have no idea how to do hand a callback to the c library.
@manveru can you please unblock me in this regard?

```
❯ nix build
error: builder for '/nix/store/xipkap5360wrljcvzfscjvim8z35ysnb-bitte-ci-0.1.1.drv' failed with exit code 1;
       last 10 log lines:
       > configuring
       > no configure script, doing nothing
       > building
       > Showing last frame. Use --error-trace for full trace.
       >
       > In src/bitte_ci/libgit2.cr:182:5
       >
       >  182 | credentials : CredentialAcquireCb
       >        ^
       > Error: field 'credentials' of struct LibGit::RemoteCallbacks has type LibGit::CredentialAcquireCb, not LibGit::Credential
       For full logs, run 'nix log /nix/store/xipkap5360wrljcvzfscjvim8z35ysnb-bitte-ci-0.1.1.drv'.
```